packages/core/nano/src/types.ts

```diff
@@ -78,17 +82,19 @@ export interface IMatrixConsumer<T> {
     cellsReplaced(row: number, col: number, numRows: number, numCols: number, values: T[], producer: IMatrixProducer<T>): void;
 }
 
-/** Provides more efficient access to 2D data for vector-aware consumers. */
+export interface IMatrixReader<T> {
+    readonly numRows: number;
+    readonly numCols: number;
+    read(row: number, col: number): T;
+}
+
+/** Provides more efficient access to 2D data for matrix-aware consumers. */
 export interface IMatrixProducer<T> {
     /**
      * Unsubscribes a consumer from this producer.
      * @param consumer - The consumer to unregister from the Producer.
      */
     removeMatrixConsumer(consumer: IMatrixConsumer<T>): void;
 
-    openMatrix(consumer: IMatrixConsumer<T>): {
-        readonly numRows: number;
-        readonly numCols: number;
-        read(row: number, col: number): T;
-    }
+    openMatrix(consumer: IMatrixConsumer<T>): IMatrixReader<T>;
 }
```

Comment:
(some unrelated interface tweaks based on experiments with IProducer/IConsumer ... should keep these, provided we keep the interfaces.)